src/testModel.ts

```diff
@@ -258,9 +320,145 @@ export class TestModel {
   }
 }
 
+export class TestModelCollection extends DisposableBase {
+  private _models: TestModel[] = [];
+  private _selectedConfigFile: string | undefined;
+  private _didUpdate: vscodeTypes.EventEmitter<void>;
+  readonly onUpdated: vscodeTypes.Event<void>;
+  private _settingsModel: SettingsModel;
+
+  constructor(vscode: vscodeTypes.VSCode, settingsModel: SettingsModel) {
+    super();
+    this._settingsModel = settingsModel;
+    this._didUpdate = new vscode.EventEmitter();
+    this.onUpdated = this._didUpdate.event;
+  }
+
+  setModelEnabled(configFile: string, enabled: boolean) {
+    const model = this._models.find(m => m.config.configFile === configFile);
+    if (!model)
+      return;
+    if (model.isEnabled === enabled)
+      return;
+    model.isEnabled = enabled;
+    this._saveSettings();
+    model.reset();
+    this._loadModelIfNeeded(model).then(() => this._didUpdate.fire());
+  }
+
+  setProjectEnabled(configFile: string, name: string, enabled: boolean) {
+    const model = this._models.find(m => m.config.configFile === configFile);
+    if (!model)
+      return;
+    const project = model.projectMap().get(name);
+    if (!project)
+      return;
+    if (project.isEnabled === enabled)
+      return;
+    project.isEnabled = enabled;
+    this._saveSettings();
+    this._didUpdate.fire();
+  }
+
+  testDirs(): string[] {
+    const result = new Set<string>();
+    for (const model of this._models) {
+      for (const dir of model.testDirs())
+        result.add(dir);
+    }
+    return [...result];
+  }
+
+  async addModel(model: TestModel) {
+    this._models.push(model);
+    const workspaceSettings = this._settingsModel.workspaceSettings.get();
+    const configSettings = (workspaceSettings.configs || []).find(c => c.relativeConfigFile === path.relative(model.config.workspaceFolder, model.config.configFile));
+    model.isEnabled = configSettings?.enabled || (this._models.length === 1 && !configSettings);
+    await this._loadModelIfNeeded(model);
+    this._disposables.push(model.onUpdated(() => this._didUpdate.fire()));
+    this._didUpdate.fire();
+  }
+
+  private async _loadModelIfNeeded(model: TestModel) {
+    if (!model.isEnabled)
+      return;
+    await model._listFiles();
+    const workspaceSettings = this._settingsModel.workspaceSettings.get();
+    const configSettings = (workspaceSettings.configs || []).find(c => c.relativeConfigFile === path.relative(model.config.workspaceFolder, model.config.configFile));
+    if (configSettings) {
+      let firstProject = true;
+      for (const project of model.projects()) {
+        const projectSettings = configSettings.projects.find(p => p.name === project.name);
+        if (projectSettings)
+          project.isEnabled = projectSettings.enabled;
+        else if (firstProject)
+          project.isEnabled = true;
+        firstProject = false;
+      }
+    } else {
+      model.projects()[0].isEnabled = true;
```

Comment:
Should we find first non-dependency project, like in UI mode?

Reply:
Good call. I did not add code / tests for dependencies yet.